### DISCUSSION TITLE
EC Storage Class data split / read performance

### DISCUSSION THREAD

**Discussion by struthio** [moosefs/moosefs#607] [Q&A]
hi,

EC storage class is available only to ARCHIVE and TRASH state, so for data that is not changing anymore.
Is EC data used in any way while reading data from MooseFS mount ?
In case of split ex. @4+1,A,B:s having B chunkservers running on slower machines impact read performance ? (*)

(*) Probably it will impact performance of moving data from CREATE/KEEP to ARCHIVE state as those parity chunks will have to be written to slower storage, that is why I ask only about read operations.

**Comment by chogata** [comment]
Data can be read from chunks in EC format, but not written to. Read performance is as good (and sometimes might be even better, with concurrent reads) as from copy format. But for writing to a chunk the chunk needs to be converted to copy format first, writing in EC format is not possible. That's why EC is only allowed for ARCHIVE state.
Of course, if you use slower machines to keep your chunks in EC format, the read operation might be slower, as much as the hardware will slow it down. But from operational point of view, for MooseFS, for reading it doesn't really matter if the chunk is in copy format or EC format.
